Getting Started adjustments, TLC...

milljm opened the pull request:
Modify most of the install instructions for some TCL.

Generalize the Manual Install Instructions a bit to be more of a reference and less "copy/paste". The goal is to move ownership away from mooseframework.org and more towards each library's respected site.

Closes #23731

Commit:
WIP Manual install instructions

Update the manual install instructions.

Closes #23731

Commit:
WIP: updates to Manual Install

The manual installation instructions needed some love.

Closes #23731

Commit:
More TLC changes

Rename "Install MOOSE" to "Build MOOSE". Makes more sense I think.
I open to debate on this.

Generalize the prereqs for all OS types.

Commit:
Remove manual_environment.md

This file is no longer being sourced.

Commit:
More good stuff

Remove cluster.md as it will no longer be sourced. The instructions
within were pretty dated, as well as something we should not be supporting.

Template where we can (more to come perhaps).

Commit:
Template INL's HPC vs HPC instructions

INL's HPC and HPC instructions should be similar.
TODO: I am not sure I like the idea of instructing folks to use
      /scratch. This location has garbage collection.

GiudGiud commented on modules/doc/content/getting_started/installation/manual_gcc.md:
```diff
@@ -1,64 +1,40 @@
 
 !package! code max-height=450
-cd $STACK_SRC/gcc-__GCC__
+cd gcc-10.4.0
```
do you want to remove the variable? seems you are adding more text variables in this PR

GiudGiud commented on modules/doc/content/getting_started/installation/manual_gcc.md:
```diff
@@ -1,64 +1,40 @@
-We need a modern C++17 capable compiler. Our minimum requirements are: GCC [!package!minimum_gcc], Clang [!package!minimum_clang]. This section will focus on building a GCC [!package!gcc] compiler stack.
+We need a modern C++17 capable compiler. Our minimum requirements are: GCC [!package!minimum_gcc],
+Clang [!package!minimum_clang]. This section will focus on building a GCC [!package!gcc] compiler
+stack.
```
```suggestion
stack. Please skip if you already have a GCC compiler of a sufficiently new version.
```

GiudGiud commented on modules/doc/content/getting_started/installation/manual_installation_gcc.md:
````diff
@@ -1,58 +1,17 @@
-
-Now that everything has been installed, its time to wrap all these environment variables up, and throw them in a bash shell profile somewhere.
-
-Append the following contents into a new file called `moose-environment.sh`:
-
-!package! code
-#!/bin/bash
-### MOOSE Environment Profile
-# GCC __GCC__
-# MPICH __MPICH__
-
-export PACKAGES_DIR=<what ever you exported initially during the Environment setup>
-
-export PATH=$PACKAGES_DIR/gcc-__GCC__/bin:$PACKAGES_DIR/mpich-__MPICH__/bin:$PACKAGES_DIR/miniconda/bin:$PATH
-export LD_LIBRARY_PATH=$PACKAGES_DIR/gcc-__GCC__/lib64:$PACKAGES_DIR/gcc-__GCC__/lib:$PACKAGES_DIR/gcc-__GCC__/lib/gcc/x86_64-pc-linux-gnu/__GCC__:$PACKAGES_DIR/gcc-__GCC__/libexec/gcc/x86_64-pc-linux-gnu/__GCC__:$PACKAGES_DIR/mpich-__MPICH__/lib:$LD_LIBRARY_PATH
-export C_INCLUDE_PATH=$PACKAGES_DIR/mpich-__MPICH__/include:$C_INCLUDE_PATH
-export CPLUS_INCLUDE_PATH=$PACKAGES_DIR/mpich-__MPICH__/include:$CPLUS_INCLUDE_PATH
-export FPATH=$PACKAGES_DIR/mpich-__MPICH__/include:$FPATH
-export MANPATH=$PACKAGES_DIR/mpich-__MPICH__/share/man:$MANPATH
-export CC=mpicc
-export CXX=mpicxx
-export FC=mpif90
-export F90=mpif90
-mamba activate peacock  # if you choose to optionally install Peacock
-!package-end!
-
-Thats it! Now you can either source this file manually each time you need to work on a MOOSE based
-application:
-
-```bash
-source /path/to/moose-environment.sh
-```
-
-Or you can permanently have it loaded each time you open a terminal by adding the above `source`
-command in your ~/.bash_profile (or ~/.bashrc which ever your system uses).
````
where is this advice now?

milljm replied:
@cticenhour @loganharbour Do you have any suggestions on this?

loganharbour replied:
Nuke it. We put a lot of effort into providing a stable stack for various kinds of users. And that will get even better when we get containers out.

These kinds of instructions should only be used now by people that get paid to manage software stacks. And those people shouldn't need these kinds of instructions.

cticenhour replied:
I agree with Logan's take. Let's do away with it.

GiudGiud commented on modules/doc/content/getting_started/installation/manual_installation_llvm.md:
```diff
@@ -1,79 +1,25 @@
 
-!include sqa/minimum_requirements.md
+Because we need MPICH to ultimately wrap to a Fortran compiler, it will be necessary to build GCC
+first, and then LLVM Clang. Therefor both compilers must be built *perfectly* (+you *really* need to
```
```suggestion
first, and then LLVM Clang. Therefore both compilers must be built *perfectly* (+you *really* need to
```

GiudGiud reviewed (changes requested):
Somebody more expert will give a better review

milljm commented on modules/doc/content/getting_started/installation/offline_installation.md:
```diff
@@ -1,46 +1,37 @@
+## Prepare Directory
 
-These three libraries can all be obtained from one repository (the moose repository). First, create a top-level directory named 'offline', which will ultimately contain everything we need to transfer over to your target machine.
+These three libraries can all be obtained from one repository (the moose repository). First, create
```
Yeah... man I always disliked these instructions. They will always be wrong (the links to those PETSc contribs are always changing).

I almost think we should differ users to Containerized solutions instead. Like Docker. I'll have to think on this. I'll make the changes, but I think in the future this page will be replaced by a containerized solution.

milljm commented on modules/doc/content/getting_started/installation/hpc_install_moose.md:
```diff
@@ -1,42 +1,45 @@
-By sourcing the above file, you are now ready to begin MOOSE-based development.
+!template load file=installation/clone_moose.md.template PATH=~/projects
 
-!alert note title=+Remember to source the profile!+
-You will need to perform the above (`source $HOME/.moose_profile`) for every new terminal session for which you perform work with MOOSE. If you want this to be automatic, add the above to your `~/.bash_profile` (or `~/.bashrc` or, which ever profile you use on your system)
```
We are instructing folks to use their HPC clusters recommended methods to aquire an MPI environment (like INL's `module load use.moose moose-dev`) So there is no need to provide these anymore.

cticenhour commented on modules/doc/config.yml:
```diff
@@ -41,7 +41,7 @@ Extensions:
         menu:
             Getting Started:
-                Installing MOOSE: getting_started/installation/index.md
+                Building MOOSE: getting_started/installation/index.md
```
I disagree with saying we're telling folks how to create their own here. The vast majority of these discussions are regarding installation of system dependencies required for MOOSE usage/development, and then building MOOSE (which is installing for a user who will only be using the built code).  

I think "Installing MOOSE" achieves the clarity we're looking for with those who are seeing the site for the first time.

Commit:
Offline Instructions TLC

Keeping inline with the changes to HPC and INL HPC, modify the
offline instructions to partially match.

Address PR suggestions/comments/grammatical fixes

Commit:
Addressing syntax comments

Forgot to address these concerns in previous commit

milljm commented on modules/doc/content/getting_started/installation/offline_installation.md:
```diff
@@ -231,16 +244,23 @@ cd $MOOSE_DIR/test
-You will need to perform the above, each and every time you log into the target machine and wish to perform development on your application. To that end, creating a special profile you can source in one command is recommended.
+You will need to perform the above, each and every time you log into the target machine and wish to
+perform development on your application. To that end, creating a special profile you can source in
+one command is recommended.
```
I always seem to miss a change  request... Yeah I will add a `!alert tip` here

GiudGiud replied:
I was thinking a snippet like:
```
export CC=....
```
and how to export that to a bash script people can run

Commit:
Convert into tip

Address comments to convert a section into an annotation.

GiudGiud commented on modules/doc/content/getting_started/installation/test_moose.md:
```diff
@@ -1,9 +1,9 @@
-## Compile and Test MOOSE
+## Build and Test MOOSE
```
SHould you use the template you created ?

modules/doc/content/getting_started/installation/test_moose.md.template

milljm replied:
We can't... If I am thinking correctly when I say we may not even need `test_moose.md.template` anymore (I can't remember. But I know when I tried to use the other templates we fail; clone_moose.md.template and the other one).

I ran into template recursion issues when attempting to use templates for our basic instructions. So these continue to need not use templates (keep the originals). But as far as our Advanced instructions, the templates work.

Sorta makes me wonder if its even worth templating for just two pages.

milljm replied:
now that I am staring at the pages, yes... we should use templates!

milljm replied:
What urks me... is that the page renders in the end, just fine.

GiudGiud replied:
oh well. It's not the biggest re-use, you're removing a lot of those re-uses here anyway

milljm replied:
@cticenhour  helped me figure out the issue :) I was doing it wrong! Of course.

So now, moar templates incoming!

GiudGiud commented on modules/doc/content/getting_started/installation/offline_installation.md:
```diff
@@ -1,52 +1,50 @@
-While arguably a [better compiler](https://opensource.apple.com/source/clang/clang-23/clang/tools/clang/www/comparison.html) over traditional GCC, it is more difficult to build. Nevertheless, if you wish to make the attempt, head on over to [llvm.org](http://llvm.org/), to begin.
+!alert note title=Personal Air-Gapped Workstation
+Procuring the above on a personal air-gapped workstation will need to be your responsibility. We
+cannot instruct a means for bypassing such security.
```
not sure what you mean here.
bypass the security of air-gapping? why are we "bypassing" ?

milljm replied:
I don't want to instruct people how to create an SSH reverse tunnel. Or how to use/bring in a thumb stick into an obviously required air-gapped secure network...

They will need to figure out how to install these prerequisites on their own. We can't help them `apt-get install`, or `xcode-select`, or use App Store, etc, behind a non-internet accessing computer. This statement basically says: Contact your help desk.

milljm replied:
I can reword it a bit? Maybe something less security-like?

```bash
!alert note title=Personal Air-Gapped Workstation
Procuring the above on a machine with no network access is beyond the scope of this document. Please work with your system administrator on satisfying the above prerequisites. 
```

GiudGiud replied:
That works

milljm commented on modules/doc/content/getting_started/installation/manual_mpich.md.template:
````diff
@@ -0,0 +1,57 @@
+```
+
+If the above command returns with paths to their respected binaries, you will want to work with your
+system package manager to remove or suppress it.
````
Yeah, except the first clause:

```bash
The following instructions are for those who wish to build their own custom GCC compiler stack.
```

When you do this, you need to build an MPI wrapper based on the new compiler

Commit:
Improving the Offline Instructions

Adressing concerns with Offline Instructions.

Commit:
Some grammatical changes

Replace wordy security statement with the general "contact your
system administrator".

Other grammatical changes.

GiudGiud commented on modules/doc/content/getting_started/installation/offline_installation.md:
```diff
@@ -39,8 +39,8 @@ If you are operating on a personal Linux workstation, the easiest way to obtain
 !alert note title=Personal Air-Gapped Workstation
-Procuring the above on a personal air-gapped workstation will need to be your responsibility. We
-cannot instruct a means for bypassing such security.
+Procuring the above on a machine with no network access is beyond the scope of this document. Please
```
```suggestion
Procuring the above on a workstation with no network access is beyond the scope of this document. Please
```

HPC cluster is within right?

milljm replied:
I am not sure I understand the question.

HPC is listed on another bullet point. I tried to separate them as best I could:

#### Air-Gapped HPC Clusters

and

#### Personal Workstation

The idea was not to make these too large, or use ## to include them in the TOC (top right table of contents). But make them large enough to separate them as sections.

milljm replied:
I thought about indenting them (like we here for each section: https://mooseframework.inl.gov/help/troubleshooting.html) But we quickly run out of space :(

Sometimes I wish we would do away with the left margin!!!

GiudGiud commented on modules/doc/content/getting_started/installation/offline_installation.md:
```diff
@@ -213,12 +214,13 @@ cd ~/offline/moose/test
+
+Most tests should pass. If they don't, or you see `MAX FAILURES`, thats a problem! And it needs to
+be addressed before continuing. Please supply a report of the actual failure (scroll up a ways). For
+example the following explains little (created with `./run_tests -i always_bad`):
```
```suggestion
example the following snippet does not give the full picture (created with `./run_tests -i always_bad`):
```

GiudGiud commented on modules/doc/content/getting_started/installation/offline_installation.md:
````diff
@@ -230,7 +232,7 @@ Ran 2 tests in 0.2 seconds. Average test time 0.0 seconds, maximum test time 0.0
 ```
 
-You need to scroll up, and find/report the actual error:
+Instead, you need to scroll up and report the actual error:
````
thanks for including this!

milljm replied:
Made this into a template. 👍

GiudGiud commented on modules/doc/content/getting_started/installation/offline_installation.md:
```diff
@@ -253,19 +255,20 @@ tests/test_harness.always_bad .................................. FAILED (CODE 1)
+For this reason you may wish to add the above export commands to your shell startup profile so that
+these variables are always set. Achieving this is different for each shell type (and there are a
+lot). You'll want to read up on how to do this for Macintosh (usually ZSH: `~/.zshrc`), or Linux
+(usually BASH: `~/.bashrc`) respectfully.
```
```suggestion
(usually BASH: `~/.bashrc`) respectively.
```

Commit:
template fixes

Addressed grammatical concerns.

Added "New User" templates to be used with the advanced instructions.

Commit:
Add styles for beautification

Adding !style! to Header paragraphs to create a bit more top margin.

Template the ~/.bashrc howto.

Commit:
Template the "New User" experience

Try and make the "New User" experience tailored to their installation
choice.

Modify the Offline instructions to use ~/projects on the Target machine.
I do not see a reason to force folks to use a top level directory named
"offline". This allowed for many existing instructions/examples to work.

Commit:
Small style changes

I feel I've made enough changes :). Time to fix the other apps.

Refs #23731

milljm commented:
This is ready, but need to hold off, until I get the failing application PR ready.

Commit:
Make New Users pagination optional

This should disconnect the examples with basic install instructions.

Commit:
Remove prompt from copyable instructions

The "copy" button is problematic when including prompt examples.

Commit:
missed a section with prompt inclusion

Commit:
Make example_failing optional

Allow the new example_failing_test markdown as an optional include.

Closes #23731

milljm commented:
This PR breaks Bison and Griffin documentation. For which I have a PR ready to go for them. Please merge when ready!

Commit:
Always more grammatical changes

Commit:
Remove unwanted mention of MPI wrappers

This is a Conda install. No need to mention MPI wrappers...

Commit:
I prefer Staying up to Date later in the doc

Placing instructions on how to stay up to date after they have built
their application feels more correct.

Commit:
Add a tip about system MPI/variables

Usually only needed on non HPC Cluster machines, I felt it cleaner
to throw a tip in the Personal Workstation section.

This might be a larger annotation than some like.

milljm commented:
ok ok ok ok... i'll stop. done for the weekend.
